src/components/editor/ResourceTemplateForm.jsx

```diff
@@ -32,21 +32,53 @@ export class ResourceTemplateForm extends Component {
     this.fulfillRTPromises(this.resourceTemplatePromise(this.joinedRTs()))
   }
 
+  /*
+   * For each fulfillled RT retrieval promise,
+   *   - dispatch loaded template to redux to be put into store
+   *   - add retrieved RT onto "nestedResourceTemplates" array in state
+   */
   fulfillRTPromises = async (promiseAll) => {
     await promiseAll.then(async (rts) => {
-      rts.map((fulfilledResourceTemplateRequest) => {
-        const joinedRts = [...this.state.nestedResourceTemplates]
+      const joinedRts = [...this.state.nestedResourceTemplates]
 
+      rts.map((fulfilledResourceTemplateRequest) => {
         joinedRts.push(fulfilledResourceTemplateRequest.response.body)
         // Add the resource template into the store
         store.dispatch(resourceTemplateLoaded(fulfilledResourceTemplateRequest.response.body))
```

Comment:
Not blocking but any particular reason this isn't part of the `mapDispatchToProps`?

Reply:
not sure.  i also found that slightly odd, but i didn't bother bringing it up during this PR, since that wasn't really the main point of it.  but i'd be happy to see this switch to using `mapDispatchToProps`, and we have some follow-on work to do to actually get #693 done, so we could maybe iron that out in the follow-on work, if you don't want to block this PR over it?

Reply:
added a comment to #693 so we remember to address this in future PR, if it isn't done here.